internal/config/config.go

```diff
@@ -241,9 +241,24 @@ type OpenStackConfig struct {
 	//   Floating IP pool to use for the VMs. For details see: https://docs.openstack.org/ocata/user-guide/cli-manage-ip-addresses.html
 	FloatingIPPoolID string `yaml:"floatingIPPoolID" validate:"required"`
 	// description: |
+	// AuthURL is the OpenStack Identity endpoint to use inside the cluster.
+	AuthURL string `yaml:"authURL" validate:"required"`
+	// description: |
+	//   ProjectID is the ID of the project where a user resides.
+	ProjectID string `yaml:"projectID" validate:"required"`
+	// description: |
+	//   ProjectName is the name of the project where a user resides.
+	ProjectName string `yaml:"projectName" validate:"required"`
+	// description: |
```

Comment:
In general, we have three sources of csp credentials:

- implicit credentials on the devops laptop used by terraform
- permissions attached to VMs (this is faked using imds on openstack but will be improved later)
- In-cluster credentials that are passed during constellation init

For the in-cluster credentials, we use a value that is retrieved from the constellation CLI and send during constellation init. This value is then stored as a kubernetes secret.

The code used to create the kubernetes secret can use the creds passed via the init call but cannot use the cloud metadata api client. For all other cloud providers this would not make sense since any credentials that can be retrieved from that API could also be retrieved from within the cloud controller manager (or similar deployments) using the cloudprovider sdk directly.
This form of authentication is only used while we are waiting for the better authentication flow so it will change soon.